### DISCUSSION TITLE
How to populate form with custom data when user clicks `Add new` button inside Join field

### DISCUSSION THREAD

**Discussion by HriBB** [payloadcms/payload#9051] [Q&A]
I want to be able to override/extend [initialDrawerState](https://github.com/payloadcms/payload/blob/beta/packages/ui/src/fields/Join/index.tsx#L60) for Join field via Payload config. Is this possible or do I need to create a custom component for that? BTW there is another issue with [RelationshipTable](https://github.com/payloadcms/payload/discussions/9049)

Video of what I am trying to achieve below. When user clicks on `Add new` button, I want to preselect the right Location (and maybe some other fields in the future). How can I do this?

[recording.webm](https://github.com/user-attachments/assets/e0f760cb-c179-4732-bde1-adc862865ef3)

**Comment by parsadotsh** [comment]
Did you find a solution to this?

**Reply by HriBB** [reply]
No, not yet. I'm waiting for some UI docs ;)
